### DISCUSSION TITLE
Stats for accept/preferred language

### DISCUSSION THREAD

**Discussion by sonnyp** [plausible/analytics#865] [Anything Else]
While stats for country can help - they do not reflect the user language and depending on the website can be misleading.

It would be great to have this stats to prioritize translations.

**Comment by ukutaht** [comment]
Yes I believe the browser has an API to get the user's locale. We should be able to use that.

**Reply by RAlfoeldi** [reply]
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/language and maybe even
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages

:-)

This would be extremely important. In countries like Switzerland we have three (actually four) relevant languages: de, it, fr and en. With only a location I don't have any chance at guessing the language.

In non-Western countries I would expect a local language as default with a fallback to English. Knowing that local language in combination with the fallbacks would also be extremely interesting. That's why stats for languages (plural) would be great!

**Comment by itadventurer** [comment]
I have implemented a very easy workaround using the [Events API](https://plausible.io/docs/custom-event-goals)

Just add this to your site:
```html
<script>
window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) };
plausible('language', {props: {languages: navigator.languages.toString()}});
</script>
```

Whenever someone visits the site, a new event is triggered. You need to add a goal `language` in your plausible configuration to see the events.

It's not pretty, but it works ;)

**Reply by RAlfoeldi** [reply]
Cool. Works. Is on production :-)

But... an integrated solution would be nice :-)

**Reply by metmarkosaric** [reply]
nice one! thanks for sharing @itadventurer and thanks for verifying @RAlfoeldi!

**Reply by RAlfoeldi** [reply]
Yeah... and it has the nice side effect of doubling the amount of hits and my monthly payment to you :-)

**Reply by RAlfoeldi** [reply]
This is what it looks like in reality. Its ok for now, but something more refined would be cool.

![Screenshot 2022-02-17 at 17 27 24](https://user-images.githubusercontent.com/7020711/154525771-584c9ee4-e57f-461e-b876-08312f0cafed.jpg)

**Reply by metmarkosaric** [reply]
yeah could be nice to add something like this to devices or locations report at some point. it's not a highly requested feature so i cannot make any promise on this at the moment. more popular requests such as the GA import take all of our time at the moment

**Reply by sandebert** [reply]
FWIW, I think it makes a lot more sense to put the locale in the device report.

**Reply by cyann** [reply]
I'm using this to make it more usable by getting only the preferred language:

```js
plausible('language',{props:{language:navigator.language.toString().substr(0,2).toLowerCase()}});
```

This keeps only the fist 2 letters (e.g. `de`) without the country code, which improves privacy.

**Comment by vanderlindenjc** [comment]
This feature would be extremely important to us, we deal with a lot of hospitality websites in Europe.

We use the navigator language to encourage our users to translate their website. You would be surprised how useful it can be to track the diversity of languages actually used on the web when you exist the "english speaking tech bubble"

**Reply by RAlfoeldi** [reply]
This is part of my main.ts

```typescript
declare let plausible:Function;

from(navigator.languages).pipe(
  map((locale) => locale.split("-")[0]),
  distinct(),
  reduce((prev, curr) => prev + "," + curr)
).subscribe((languages) => {
  plausible("language", {props: {languages: languages}})
});
```

and gives me the stats I mentioned above. It doesn't give you a full statistical analysis, but a pretty good idea of what languages your users understand.

**Comment by paulschreiber** [comment]
It would be most helpful if every page view/event included the locale/languages.

**Comment by nicbou** [comment]
This is more relevant to me than the browser information. As the users above, I want to prioritize translations and cater to specific cultures. It's important data, but I don't want to pay twice as much to track this extra dimension.
